core/src/main/java/jenkins/model/ParameterizedJobMixIn.java

```diff
@@ -238,12 +239,12 @@ public final void doBuildWithParameters(StaplerRequest2 req, StaplerResponse2 rs
 
         ParametersDefinitionProperty pp = asJob().getProperty(ParametersDefinitionProperty.class);
         if (!asJob().isBuildable()) {
-            throw HttpResponses.error(SC_CONFLICT, new IOException(asJob().getFullName() + " is not buildable!"));
+            throw HttpResponses.errorWithoutStack(SC_CONFLICT, asJob().getFullName() + " is not buildable");
         }
         if (pp != null) {
             pp.buildWithParameters(req, rsp, delay);
         } else {
-            throw new IllegalStateException("This build is not parameterized!");
+            throw HttpResponses.errorWithoutStack(SC_BAD_REQUEST, asJob().getFullName() + " is not parameterized");
```

Comment:
Response also did not specify the job (though I guess it should be obvious from the URL), and said _build_ when _job_ (or _project_) was meant.